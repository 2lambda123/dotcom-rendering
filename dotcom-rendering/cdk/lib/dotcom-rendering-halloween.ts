import type { Alarms } from '@guardian/cdk';
import { GuNodeApp } from '@guardian/cdk';
import { AccessScope } from '@guardian/cdk/lib/constants';
import type { NoMonitoring } from '@guardian/cdk/lib/constructs/cloudwatch';
import { GuStack, type GuStackProps } from '@guardian/cdk/lib/constructs/core';
import type { GuAsgCapacity } from '@guardian/cdk/lib/types';
import type { App } from 'aws-cdk-lib';
import { Monitoring } from 'aws-cdk-lib/aws-autoscaling';
import {
	InstanceClass,
	type InstanceSize,
	InstanceType,
	IpAddresses,
	Peer,
} from 'aws-cdk-lib/aws-ec2';
import { satisfies } from 'compare-versions';
import { getUserData } from './userData';

interface DCRHalloweenProps extends GuStackProps {
	app: string;
	instanceSize: InstanceSize;
	scaling: GuAsgCapacity;
}

export class DotcomRenderingHalloween extends GuStack {
	// This should be coming from the base class, no?
	constructor(scope: App, id: string, props: DCRHalloweenProps) {
		super(scope, id, props);

		const { region } = this;
		const { app, stage, instanceSize, scaling } = props;

		const userData = getUserData({
			app,
			elkStreamId: 'fake-value-for-now-please-change-me',
			region,
			stage,
		});

		const vpcCidrBlock = '10.248.136.0/22';
		// const vpc = GuVpc.fromIdParameter(this, 'vpc', { vpcCidrBlock });
		// const publicSubnets = GuVpc.subnetsFromParameter(this, {
		// 	type: SubnetType.PUBLIC,
		// });
		// const privateSubnets = GuVpc.subnetsFromParameter(this, {
		// 	type: SubnetType.PRIVATE,
		// });

		// const criticalAlertsTopicArn = `arn:aws:sns:${region}:${this.account}:Frontend-${stage}-CriticalAlerts`;

		const monitoringConfiguration =
			stage === 'PROD'
				? ({
						snsTopicName: `Frontend-${stage}-CriticalAlerts`,
						// TODO - how does this overlap with the DevX debug dashboard?
						http5xxAlarm: {
							tolerated5xxPercentage: 10,
							numberOfMinutesAboveThresholdBeforeAlarm: 1,
						},
						unhealthyInstancesAlarm: true,
				  } satisfies Alarms)
				: ({ noMonitoring: true } satisfies NoMonitoring);

		const nodeApp = new GuNodeApp(this, {
			// access: '',
			app,
			access: {
				// TODO - ask DevX about Peer & CIDR ranges (and VPCs)
				// … and what would be the meaning of an empty array?
				// Should it be [IPeer, …IPeer[]]
				cidrRanges: [Peer.ipv4(vpcCidrBlock)],
				scope: AccessScope.INTERNAL,
			},
			certificateProps: { domainName: 'www.theguardian.com' },
			instanceType: InstanceType.of(InstanceClass.T4G, instanceSize),
			monitoringConfiguration,
			// { minimumInstances: props.minCapacity, maximumInstances: props.maxCapacity}
			scaling,
			userData,
		});

		console.log(nodeApp);
	}
}
