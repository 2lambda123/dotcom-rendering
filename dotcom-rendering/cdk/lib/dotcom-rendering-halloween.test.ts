import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { InstanceSize } from 'aws-cdk-lib/aws-ec2';
import { DotcomRenderingHalloween } from './dotcom-rendering-halloween';

/**
 * These tests make sure that the cloudformation template being generated by CDK has not deviated
 * from the cloudformation snapshot that has been checked into our repository.
 */
describe('The DotcomRendering stack', () => {
	it('matches the snapshot', () => {
		const app = new App();
		const stack = new DotcomRenderingHalloween(app, 'DotcomRendering', {
			stack: 'frontend',
			stage: 'TEST',
			app: 'rendering',
			instanceSize: InstanceSize.MICRO,
			scaling: { minimumInstances: 1, maximumInstances: 4 },
		});
		const template = Template.fromStack(stack);
		expect(template.toJSON()).toMatchSnapshot();
	});
});
