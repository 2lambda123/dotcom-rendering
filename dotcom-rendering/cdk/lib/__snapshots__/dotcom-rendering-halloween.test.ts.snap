// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The DotcomRendering stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuVpcParameter",
      "GuSubnetListParameter",
      "GuSubnetListParameter",
      "GuNodeApp",
      "GuCertificate",
      "GuInstanceRole",
      "GuDescribeEC2Policy",
      "GuLoggingStreamNameParameter",
      "GuLogShippingPolicy",
      "GuDistributionBucketParameter",
      "GuGetDistributablePolicy",
      "GuParameterStoreReadPolicy",
      "GuAmiParameter",
      "GuHttpsEgressSecurityGroup",
      "GuWazuhAccess",
      "GuAutoScalingGroup",
      "GuApplicationLoadBalancer",
      "GuApplicationTargetGroup",
      "GuHttpsApplicationListener",
      "GuSecurityGroup",
    ],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "LoadBalancerRenderingDnsName": {
      "Description": "DNS entry for LoadBalancerRendering",
      "Value": {
        "Fn::GetAtt": [
          "LoadBalancerRendering3F722A50",
          "DNSName",
        ],
      },
    },
  },
  "Parameters": {
    "AMIRendering": {
      "Description": "Amazon Machine Image ID for the app rendering. Use this in conjunction with AMIgo to keep AMIs up to date.",
      "Type": "AWS::EC2::Image::Id",
    },
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "LoggingStreamName": {
      "Default": "/account/services/logging.stream.name",
      "Description": "SSM parameter containing the Name (not ARN) on the kinesis stream",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "VpcId": {
      "Default": "/account/vpc/primary/id",
      "Description": "Virtual Private Cloud to run EC2 instances within. Should NOT be the account default VPC.",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::VPC::Id>",
    },
    "renderingPrivateSubnets": {
      "Default": "/account/vpc/primary/subnets/private",
      "Description": "A list of private subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
    "renderingPublicSubnets": {
      "Default": "/account/vpc/primary/subnets/public",
      "Description": "A list of public subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
  },
  "Resources": {
    "AutoScalingGroupRenderingASG3C2E86A8": {
      "Properties": {
        "HealthCheckGracePeriod": 120,
        "HealthCheckType": "ELB",
        "LaunchTemplate": {
          "LaunchTemplateId": {
            "Ref": "frontendTESTrenderingF3FD9600",
          },
          "Version": {
            "Fn::GetAtt": [
              "frontendTESTrenderingF3FD9600",
              "LatestVersionNumber",
            ],
          },
        },
        "MaxSize": "4",
        "MinSize": "1",
        "Tags": [
          {
            "Key": "App",
            "PropagateAtLaunch": true,
            "Value": "rendering",
          },
          {
            "Key": "gu:cdk:version",
            "PropagateAtLaunch": true,
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "PropagateAtLaunch": true,
            "Value": "guardian/dotcom-rendering",
          },
          {
            "Key": "LogKinesisStreamName",
            "PropagateAtLaunch": true,
            "Value": {
              "Ref": "LoggingStreamName",
            },
          },
          {
            "Key": "Stack",
            "PropagateAtLaunch": true,
            "Value": "frontend",
          },
          {
            "Key": "Stage",
            "PropagateAtLaunch": true,
            "Value": "TEST",
          },
        ],
        "TargetGroupARNs": [
          {
            "Ref": "TargetGroupRendering097B53D5",
          },
        ],
        "VPCZoneIdentifier": {
          "Ref": "renderingPrivateSubnets",
        },
      },
      "Type": "AWS::AutoScaling::AutoScalingGroup",
    },
    "CertificateRendering3CD8FCA4": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "DomainName": "www.theguardian.com",
        "Tags": [
          {
            "Key": "App",
            "Value": "rendering",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/dotcom-rendering",
          },
          {
            "Key": "Name",
            "Value": "DotcomRendering/CertificateRendering",
          },
          {
            "Key": "Stack",
            "Value": "frontend",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "ValidationMethod": "DNS",
      },
      "Type": "AWS::CertificateManager::Certificate",
      "UpdateReplacePolicy": "Retain",
    },
    "DescribeEC2PolicyFF5F9295": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "autoscaling:DescribeAutoScalingInstances",
                "autoscaling:DescribeAutoScalingGroups",
                "ec2:DescribeTags",
                "ec2:DescribeInstances",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "describe-ec2-policy",
        "Roles": [
          {
            "Ref": "InstanceRoleRenderingB06D827E",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GetDistributablePolicyRenderingCA2FF5C7": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:s3:::",
                    {
                      "Ref": "DistributionBucketName",
                    },
                    "/frontend/TEST/rendering/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GetDistributablePolicyRenderingCA2FF5C7",
        "Roles": [
          {
            "Ref": "InstanceRoleRenderingB06D827E",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GuHttpsEgressSecurityGroupRenderingE17B2710": {
      "Properties": {
        "GroupDescription": "Allow all outbound HTTPS traffic",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound HTTPS traffic",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "rendering",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/dotcom-rendering",
          },
          {
            "Key": "Stack",
            "Value": "frontend",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "GuHttpsEgressSecurityGroupRenderingfromDotcomRenderingInternalIngressSecurityGroupRendering646802763000EBB32877": {
      "Properties": {
        "Description": "Load balancer to target",
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "GuHttpsEgressSecurityGroupRenderingE17B2710",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "InternalIngressSecurityGroupRendering49552846",
            "GroupId",
          ],
        },
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "GuHttpsEgressSecurityGroupRenderingfromDotcomRenderingLoadBalancerRenderingSecurityGroup6F454DD2300060A2C747": {
      "Properties": {
        "Description": "Load balancer to target",
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "GuHttpsEgressSecurityGroupRenderingE17B2710",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "LoadBalancerRenderingSecurityGroup0072305D",
            "GroupId",
          ],
        },
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "GuLogShippingPolicy981BFE5A": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "kinesis:Describe*",
                "kinesis:Put*",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:kinesis:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":stream/",
                    {
                      "Ref": "LoggingStreamName",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GuLogShippingPolicy981BFE5A",
        "Roles": [
          {
            "Ref": "InstanceRoleRenderingB06D827E",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "InstanceRoleRenderingB06D827E": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "ec2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AmazonSSMManagedInstanceCore",
              ],
            ],
          },
        ],
        "Path": "/",
        "Tags": [
          {
            "Key": "App",
            "Value": "rendering",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/dotcom-rendering",
          },
          {
            "Key": "Stack",
            "Value": "frontend",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "InternalIngressSecurityGroupRendering49552846": {
      "Properties": {
        "GroupDescription": "Allow restricted ingress from CIDR ranges",
        "SecurityGroupIngress": [
          {
            "CidrIp": "10.248.136.0/22",
            "Description": "Allow access on port 443 from 10.248.136.0/22",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "rendering",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/dotcom-rendering",
          },
          {
            "Key": "Stack",
            "Value": "frontend",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "InternalIngressSecurityGroupRenderingtoDotcomRenderingGuHttpsEgressSecurityGroupRendering163CC70B30009B55A026": {
      "Properties": {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "GuHttpsEgressSecurityGroupRenderingE17B2710",
            "GroupId",
          ],
        },
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "InternalIngressSecurityGroupRendering49552846",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "InternalIngressSecurityGroupRenderingtoDotcomRenderingWazuhSecurityGroupFB3C43813000E4CECC37": {
      "Properties": {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "WazuhSecurityGroup",
            "GroupId",
          ],
        },
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "InternalIngressSecurityGroupRendering49552846",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "ListenerRendering2ED60F11": {
      "Properties": {
        "Certificates": [
          {
            "CertificateArn": {
              "Ref": "CertificateRendering3CD8FCA4",
            },
          },
        ],
        "DefaultActions": [
          {
            "TargetGroupArn": {
              "Ref": "TargetGroupRendering097B53D5",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": {
          "Ref": "LoadBalancerRendering3F722A50",
        },
        "Port": 443,
        "Protocol": "HTTPS",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "LoadBalancerRendering3F722A50": {
      "Properties": {
        "LoadBalancerAttributes": [
          {
            "Key": "deletion_protection.enabled",
            "Value": "true",
          },
        ],
        "Scheme": "internal",
        "SecurityGroups": [
          {
            "Fn::GetAtt": [
              "LoadBalancerRenderingSecurityGroup0072305D",
              "GroupId",
            ],
          },
          {
            "Fn::GetAtt": [
              "InternalIngressSecurityGroupRendering49552846",
              "GroupId",
            ],
          },
        ],
        "Subnets": {
          "Ref": "renderingPrivateSubnets",
        },
        "Tags": [
          {
            "Key": "App",
            "Value": "rendering",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/dotcom-rendering",
          },
          {
            "Key": "Stack",
            "Value": "frontend",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "Type": "application",
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "LoadBalancerRenderingSecurityGroup0072305D": {
      "Properties": {
        "GroupDescription": "Automatically created Security Group for ELB DotcomRenderingLoadBalancerRendering5BEB2BF1",
        "Tags": [
          {
            "Key": "App",
            "Value": "rendering",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/dotcom-rendering",
          },
          {
            "Key": "Stack",
            "Value": "frontend",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "LoadBalancerRenderingSecurityGrouptoDotcomRenderingGuHttpsEgressSecurityGroupRendering163CC70B30002BE07DEE": {
      "Properties": {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "GuHttpsEgressSecurityGroupRenderingE17B2710",
            "GroupId",
          ],
        },
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "LoadBalancerRenderingSecurityGroup0072305D",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "LoadBalancerRenderingSecurityGrouptoDotcomRenderingWazuhSecurityGroupFB3C43813000F901AB79": {
      "Properties": {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "WazuhSecurityGroup",
            "GroupId",
          ],
        },
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "LoadBalancerRenderingSecurityGroup0072305D",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "ParameterStoreReadRenderingFDC91AAA": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/frontend/rendering",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/frontend/rendering/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "parameter-store-read-policy",
        "Roles": [
          {
            "Ref": "InstanceRoleRenderingB06D827E",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TargetGroupRendering097B53D5": {
      "Properties": {
        "HealthCheckIntervalSeconds": 10,
        "HealthCheckPath": "/healthcheck",
        "HealthCheckProtocol": "HTTP",
        "HealthCheckTimeoutSeconds": 5,
        "HealthyThresholdCount": 5,
        "Port": 3000,
        "Protocol": "HTTP",
        "Tags": [
          {
            "Key": "App",
            "Value": "rendering",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/dotcom-rendering",
          },
          {
            "Key": "Stack",
            "Value": "frontend",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "TargetGroupAttributes": [
          {
            "Key": "deregistration_delay.timeout_seconds",
            "Value": "30",
          },
          {
            "Key": "stickiness.enabled",
            "Value": "false",
          },
        ],
        "TargetType": "instance",
        "UnhealthyThresholdCount": 2,
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
    "WazuhSecurityGroup": {
      "Properties": {
        "GroupDescription": "Allow outbound traffic from wazuh agent to manager",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh event logging",
            "FromPort": 1514,
            "IpProtocol": "tcp",
            "ToPort": 1514,
          },
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh agent registration",
            "FromPort": 1515,
            "IpProtocol": "tcp",
            "ToPort": 1515,
          },
        ],
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/dotcom-rendering",
          },
          {
            "Key": "Stack",
            "Value": "frontend",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "WazuhSecurityGroupfromDotcomRenderingInternalIngressSecurityGroupRendering646802763000E8CC9B9C": {
      "Properties": {
        "Description": "Load balancer to target",
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "WazuhSecurityGroup",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "InternalIngressSecurityGroupRendering49552846",
            "GroupId",
          ],
        },
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "WazuhSecurityGroupfromDotcomRenderingLoadBalancerRenderingSecurityGroup6F454DD23000AABF575D": {
      "Properties": {
        "Description": "Load balancer to target",
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "WazuhSecurityGroup",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "LoadBalancerRenderingSecurityGroup0072305D",
            "GroupId",
          ],
        },
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "frontendTESTrenderingF3FD9600": {
      "Properties": {
        "LaunchTemplateData": {
          "IamInstanceProfile": {
            "Arn": {
              "Fn::GetAtt": [
                "frontendTESTrenderingProfileA2D2425E",
                "Arn",
              ],
            },
          },
          "ImageId": {
            "Ref": "AMIRendering",
          },
          "InstanceType": "t4g.micro",
          "MetadataOptions": {
            "HttpTokens": "required",
          },
          "SecurityGroupIds": [
            {
              "Fn::GetAtt": [
                "GuHttpsEgressSecurityGroupRenderingE17B2710",
                "GroupId",
              ],
            },
            {
              "Fn::GetAtt": [
                "WazuhSecurityGroup",
                "GroupId",
              ],
            },
          ],
          "TagSpecifications": [
            {
              "ResourceType": "instance",
              "Tags": [
                {
                  "Key": "gu:cdk:version",
                  "Value": "TEST",
                },
                {
                  "Key": "gu:repo",
                  "Value": "guardian/dotcom-rendering",
                },
                {
                  "Key": "Name",
                  "Value": "DotcomRendering/frontend-TEST-rendering",
                },
                {
                  "Key": "Stack",
                  "Value": "frontend",
                },
                {
                  "Key": "Stage",
                  "Value": "TEST",
                },
              ],
            },
            {
              "ResourceType": "volume",
              "Tags": [
                {
                  "Key": "gu:cdk:version",
                  "Value": "TEST",
                },
                {
                  "Key": "gu:repo",
                  "Value": "guardian/dotcom-rendering",
                },
                {
                  "Key": "Name",
                  "Value": "DotcomRendering/frontend-TEST-rendering",
                },
                {
                  "Key": "Stack",
                  "Value": "frontend",
                },
                {
                  "Key": "Stage",
                  "Value": "TEST",
                },
              ],
            },
          ],
          "UserData": {
            "Fn::Base64": {
              "Fn::Join": [
                "",
                [
                  "#!/bin/bash -ev
groupadd frontend
useradd -r -m -s /usr/bin/nologin -g frontend dotcom-rendering
usermod -a -G frontend aws-kinesis-agent-user
cd /home/dotcom-rendering
aws --region eu-west-1 s3 cp s3://aws-frontend-artifacts/frontend/TEST/rendering/rendering.tar.gz ./
tar -zxf rendering.tar.gz rendering
chown -R dotcom-rendering:frontend rendering
cd rendering
export TERM=xterm-256color
export NODE_ENV=production
export GU_STAGE=TEST
mkdir /var/log/dotcom-rendering
chown -R dotcom-rendering:frontend /var/log/dotcom-rendering
sudo NODE_ENV=$NODE_ENV GU_STAGE=$GU_STAGE -u dotcom-rendering -g frontend make start-prod
/opt/aws-kinesis-agent/configure-aws-kinesis-agent ",
                  {
                    "Ref": "AWS::Region",
                  },
                  " fake-value-for-now-please-change-me /var/log/dotcom-rendering/dotcom-rendering.log",
                ],
              ],
            },
          },
        },
        "TagSpecifications": [
          {
            "ResourceType": "launch-template",
            "Tags": [
              {
                "Key": "gu:cdk:version",
                "Value": "TEST",
              },
              {
                "Key": "gu:repo",
                "Value": "guardian/dotcom-rendering",
              },
              {
                "Key": "Name",
                "Value": "DotcomRendering/frontend-TEST-rendering",
              },
              {
                "Key": "Stack",
                "Value": "frontend",
              },
              {
                "Key": "Stage",
                "Value": "TEST",
              },
            ],
          },
        ],
      },
      "Type": "AWS::EC2::LaunchTemplate",
    },
    "frontendTESTrenderingProfileA2D2425E": {
      "Properties": {
        "Roles": [
          {
            "Ref": "InstanceRoleRenderingB06D827E",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
  },
}
`;
